Sources/Components/Feed/InAppFeedViewModel.swift

```diff
@@ -0,0 +1,312 @@
+                let originalStatus = feedClientOptions.status
+                let archived: Knock.FeedItemArchivedScope? = feedClientOptions.status == .archived ? .only : nil
+                let status = feedClientOptions.status == .archived ? .all : feedClientOptions.status
+                self.feedClientOptions.archived = archived
+                self.feedClientOptions.status = status
+                self.feedClientOptions.before = nil
+                self.feedClientOptions.after = nil
+                guard let userFeed = try await Knock.shared.feedManager?.getUserFeedContent(options: feedClientOptions) else { return }
+                DispatchQueue.main.async {
+                    self.feed = userFeed
+                    self.feed.pageInfo.before = self.feed.entries.first?.__cursor
+                    self.feedClientOptions.status = originalStatus
+                    if self.showRefreshIndicator {
+                        self.showRefreshIndicator = false
+                    }
+                }
+            } catch {
+                handleFeedError(error)
+            }
+        }
+        
+        /// Fetches a new page of feed items when the user scrolls to the bottom of the feed.
+        public func fetchNewPageOfFeedItems() async {
+            guard let after = self.feed.pageInfo.after else { return }
+            feedClientOptions.after = after
+            Knock.shared.feedManager?.getUserFeedContent(options: feedClientOptions) { result in
+                DispatchQueue.main.async {
+                    switch result {
+                    case .success(let feed):
+                        self.mergeFeedsForNewPageOfFeed(feed: feed)
+                    case .failure(let error):
+                        self.handleFeedError(error)
+                    }
+                }
+            }
+        }
+        
+        /// Determines whether there is another page of content to fetch when paginaating the feedItem list
+        public func isMoreContentAvailable() -> Bool {
+            return feed.pageInfo.after != nil
+        }
+        
+        // MARK: Message Update Methods
+        
+        /// Archives a specific feed item.
+        public func archiveItem(_ item: Knock.FeedItem) async {
+            do {
+                let message = try await Knock.shared.updateMessageStatus(messageId: item.id, status: .archived)
+                
+                // remove local message if update was successful
+                DispatchQueue.main.async {
+                    self.feed.entries = self.feed.entries.filter{ $0.id != message.id }
+                }
+                await fetchNewMetaData()
+            } catch {
+                Knock.shared.log(type: .error, category: .feed, message: "error in archiveItem: \(error.localizedDescription)")
+            }
+        }
+        
+        /// Archives all items within the specified scope.
+        public func archiveAll(scope: Knock.FeedItemScope) async {
```

Comment:
One thing about all of these `xAll` functions is that the `refreshFeed` and `fetchNewMetaData` isn't always going to return the feed with the updates applied. That's because the `All` methods make a bulk update, which returns a `BulkOperation` which is **async** on the server. 

On the web side, what we do here is we maintain a **state store** of all of the notifications from the server and then make the update **optimistically**  on the client to reflect the changes (e.g. in this case, setting `archived_at` on the items) and manually updating the metadata state.